js/colourpicker.js

```diff
@@ -153,16 +143,22 @@
       if( input.data('colourpicker-initialized') ) return;
 
       // Handle settings
-      settings = $.extend(true, {}, defaults, settings);
-
-	    // Palette type
-	    colourpicker.addClass('palette-' + settings.palette);
-	  
-	    // If adding a transparent checkbox, make the parent an input group
-      if( settings.allowTransparent ) {
-        colourpicker.addClass('input-group');
+      settings = $.extend(false, {}, defaults, settings);
+      //for backwards compatibility
+      if ($.type(settings.allowedCols) === "string") {
+        settings.allowedCols = settings.allowedCols.split(" ");
       }
-
+	    // Palette type, alpha or not, make it an input group.
```

Comment:
why is it always an input-group? I wanted it to only be an input group when there is actually another UI component embedded inside, so the UI needed to look right, but when there is only one input I don't see the need?